analyze/workerd.ts

```diff
@@ -97,14 +123,22 @@ export async function isValidEmail(
   context: AnalyzeContext,
   candidate: string,
   options?: EmailValidationConfig,
-) {
+): Promise<EmailValidationResult> {
   const analyze = await init(context);
+  const optionsOrDefault = options || {
+    requireTopLevelDomain: true,
+    allowDomainLiteral: false,
+    blockedEmails: [],
+  };
```

Comment:
I think this is the incorrect way to default these. I think we need something like:
```suggestion
  const optionsOrDefault = {
    requireTopLevelDomain: true,
    allowDomainLiteral: false,
    blockedEmails: [],
    ...options,
  };
```
I believe this will both in the scenario where `options` is undefined **and** when it does not contain all properties.